src/basic/mkdir-label.c

```diff
@@ -34,6 +34,23 @@ int mkdir_label(const char *path, mode_t mode) {
         return mac_smack_fix(path, 0);
 }
 
+int mkdirat_label(int dirfd, const char *path, mode_t mode) {
+        int r;
+
+        assert(path);
+
+        r = mac_selinux_create_file_prepare(path, S_IFDIR);
```

Comment:
this is wrong, the selinux label is derived from the absolute path, and `path` in this case might be relative to dirfd...

maybe using fd_get_path() is a way out, though an ugly one

Reply:
ouch, it looks like I was hurry to terminate this work :D